admin_manual/configuration/occ_command.rst

```diff
@@ -0,0 +1,252 @@
+   
+TODO: does this count LDAP and other external users, or local only?::
+
+ $ sudo -u www-data php occ user:report
+   +------------------+---+
+   | User Report      |   |
+   +------------------+---+
+   | OC_User_Database | 2 |
+   |                  |   |
+   | total users      | 2 |
+   |                  |   |
+   | user directories | 3 |
+   +------------------+---+
+   
+Apps Commands
+-------------
+
+The ``app`` commands list, enable, and disable apps. This lists all of your 
+installed apps, and shows whether they are enabled or disabled::
+
+ $ sudo -u www-data php occ app:list
+ 
+Enable an app::
+
+ $ sudo -u www-data php occ app:enable external
+   external enabled
+   
+Disable an app::
+
+ $ sudo -u www-data php occ app:disable external
+   external disabled
+   
+Upgrade Command
+---------------
+
+When you are performing an update or upgrade on your ownCloud server, it is 
+better to use ``occ`` to perform the database upgrade step, rather than the Web 
+GUI,  in order to avoid timeouts.
+
+TODO: what timeouts? What causes them?
+
+You can perform a dry-run first to see what will happen, without changing 
+anything::
+
+ $ sudo -u www-data php occ upgrade --dry-run
+
+When this looks satisfactory, you can go ahead and perform the upgrade::
+
+ $ sudo -u www-data php occ upgrade
+ 
+TODO why would you want to use --skip-migration-test? ::
+  
+ $ sudo -u www-data php occ upgrade --skip-migration-test
+ 
+
+File Scanning
+-------------
+
+The ``files:scan`` command lists all files belonging to a specified user, and 
+all files that belong to all users. This lists all files that belong to user 
+layla::
+
+ $ sudo -u www-data php occ files:scan layla
+ 
+This lists all files owned by all of your ownCloud users::
+
+ $ sudo -u www-data php occ files:scan --all
+
+You can store all those filenames in a text file. The file must be created in a 
+directory that you have write permissions to, such as your home directory::
+
+  $ sudo -u www-data php occ files:scan layla --all > /home/user/ocfilelist.txt
+
+Database Conversion
+-------------------
+
+The SQLite database is good for testing, and for ownCloud servers with small 
+workloads, but servers with more than a few users and data files should use 
+MariaDB, MySQL, PostgreSQL, or Oracle. You can use ``occ`` to convert from 
+SQLite to one of these other databases. You need:
+
+* Your desired database installed and its PHP connector
+* The login and password of a database admin user
+* The database port number, if it is a non-standard port
+
+This is example converts to MariaDB, and also converts the schema for all 
+installed apps:: 
+
+ $ sudo -u www-data php occ db:generate-change-script
+ $ sudo -u www-data php occ db:convert-type --all-apps mysql oc_dbuser 127.0.0.1 oc_database
+
+For a more detailed explanation see :doc:`../maintenance/convert_db`   
+
+LDAP Commands
+-------------
+
+TODO: explanation of each LDAP command, and example command output
```

Comment:
## commands in OC 7

### ldap:check-user [--force] ocName      
checks whether a user exists on LDAP. It will only work, if all configured LDAP configurations are active. If you insist to run it while having inactive LDAP configurations, you can pass "--force" as parameter. The "ocName" parameter is the username as used in ownCloud. If the user was found to have been deleted on LDAP the user will be marked as such and the user's remainings can be deleted from ownCloud using user:delete command.

### ldap:search [--group] [--offset="..."] [--limit="..."] search

Performs a search for users (or groups if ```--group``` is specified) against LDAP. With offset (default 0) and limit (default 15), paging can be controlled. Offset must be a multiple of limit or 0. 

Example (output is displayname and ownCloud username in brackets):

```
$ ./occ ldap:search --offset 30 --limit 10 ro
Roberts, Margie (7a1a4da4-423b-1033-9012-ad5f418ee02e)
Rodgers, Don (79e543ca-423b-1033-8fb7-ad5f418ee02e)
Rodgers, Sadie (79cb3b38-423b-1033-8f89-ad5f418ee02e)
Rodgers, Sonia (79cbd728-423b-1033-8f8a-ad5f418ee02e)
Rogers, Christoph (7a1cd9de-423b-1033-9016-ad5f418ee02e)
Rogers, Margie (7a408a3c-423b-1033-9046-ad5f418ee02e)
Rose, Andre (7a2492aa-423b-1033-9021-ad5f418ee02e)
Rose, Julia (7af0f76e-423b-1033-9066-ad5f418ee02e)
Rose, Priscilla (79f14648-423b-1033-8fcc-ad5f418ee02e)
Rowland, Richard (7b2f9fbe-423b-1033-90c5-ad5f418ee02e)
```

### ldap:set-config configID configKey configValue

Manipulates a config entry. configID is the configuration prefix, might be "" (empty, you need to pass double quotes) for the first one, afterwards s01, s02, sXX. You can find out the prefix with ldap:show-config. configKey is the configuration name you want to modify (again, find the right one using ldap:show-config). configValue parameter is the new value for the configuration. 

Example: add more attributes for user search:

```
$ ./occ ldap:set-config s01 ldapAttributesForUserSearch "cn;givenname;sn;displayname;mail"
```

### ldap:show-config [configID]

Prints all LDAP configuration or the one specified by configID, e.g. "s01".

### ldap:show-remnants

Display all LDAP users that have been found as deleted on LDAP and marked as such. You can delete the remainings using user:delete and passing the "ownCloud name". Example output:

```
+---------------+----------------+-------------+------------------------------------------------+-------------------------+----------------------------+--------+
| ownCloud name | Display Name   | LDAP UID    | LDAP DN                                        | Last Login              | Dir                        | Sharer |
+---------------+----------------+-------------+------------------------------------------------+-------------------------+----------------------------+--------+
| zombie12839   | May, Arlene    | zombie12839 | uid=zombie12839,ou=zombies,dc=owncloud,dc=bzoc | December  8, 2014 23:03 | /home/openldap/zombie12839 | N      |
| zombie972     | Kennedy, Alice | zombie972   | uid=zombie972,ou=zombies,dc=owncloud,dc=bzoc   | December  8, 2014 23:02 | /home/openldap/zombie972   | N      |
+---------------+----------------+-------------+------------------------------------------------+-------------------------+----------------------------+--------+
```

### ldap:test-config configID

Tests whether an LDAP configuration can actually connect and bind to the LDAP server. 

```
$ ./occ ldap:test-config s01
The configuration is valid, but the Bind failed. Please check the server settings and credentials.
```

```
$ ./occ ldap:test-config ""
The configuration is valid and the connection could be established!
```

## commands in OC 8 (additional to those above)

### ldap:create-empty-config

Creates an empty LDAP configuration which can be modified later using ldpa:set-config.

```
$ ./occ ldap:create-empty-config 
Created new configuration with configID 's08'
```

### ldap:delete-config configID

Deletes an LDAP configuration.

```
$ ./occ ldap:delete-config s08
Deleted configuration with configID 's08'
```